data/sample_dataset.py

```diff
@@ -0,0 +1,28 @@
+import csv
+from random import sample
+
+lines = []
+class_0_count = 0
+with open("banknote_auth_full.txt", mode="r") as f:
+    csv_file = csv.reader(f)
+
+    max_0 = 0
+    for i, line in enumerate(csv_file):
+        if line[-1] == "0":
+            class_0_count += 1
+            max_0 = i
+        lines.append(line)
+
+class_0_percent = int(round(100 * class_0_count / len(lines)))
+class_1_percent = 100 - class_0_percent
+
+class_0_data = sample(lines[:max_0], class_0_percent)
+class_1_data = sample(lines[max_0 + 1 :], class_1_percent)
+
+all_data = class_0_data + class_1_data
+
```

Comment:
# maybe add this so that we can use the training data directly
x_data = np.array(all_data)[:,:4]
y_data = np.array(all_data)[:,-1]

Reply:
Thanks for the suggestion, I should have added a little bit more reason for this script and what I think it should be used for.

The `data/sample_dataset.py` script was used to generate 100 samples of our banknote dataset and create `data/banknote_auth_100.csv` since our models will likely not be able to consume too much data. This file should _not_ be used to start doing any machine learning. I would imagine we should have another file that loads the data, splits it into features and classes, and probably calls the [`train_test_split`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) from [scikit-learn](https://scikit-learn.org/stable/index.html).

Hence, I do not think we should add any code in this script to preemptively split the data into features/classes.

Reply:
I see. Then it's good enough!